assets/scripts/store/actions/street.js

```diff
@@ -371,3 +371,51 @@ export const getLastStreet = () => {
     }
   }
 }
+export const createNewStreet = () => {
+  return async (dispatch, getState) => {
+    const { signedIn } = getState().user
+    const defaultStreet = createEmptyStreet(getState())
+    dispatch(updateStreetData(defaultStreet))
+    if (signedIn) {
+      await dispatch(setSettings({
+        lastStreetId: defaultStreet.id,
+        lastStreetNamespacedId: defaultStreet.namespacedId,
+        lastStreetCreatorId: defaultStreet.creatorId
+      }))
+    }
+
+    const response = await apiClient.saveStreet(prepareStreetForSaving(defaultStreet))
+    dispatch(saveStreetId(response.id, response.namespacedId))
+    // refactor once API save street has been added to actions
+    saveStreetToServer(true)
+  }
+}
+const prepareStreetForSaving = (street) => {
+  const data = { street }
+  const name = street.name
+  const originalStreetId = street.originalStreetId
+
+  // Those go above data in the structure, so they need to be cleared here
+  delete data.street.name
+  delete data.street.originalStreetId
+  delete data.street.updatedAt
+
+  // This will be implied through authorization header
+  delete data.street.creatorId
+
+  /**
+   * ToDo: move this outside
+  if (store.getState().flags.SAVE_UNDO.value === true) {
+    data.undoStack = cloneDeep(store.getState().undo.stack)
+    data.undoPosition = store.getState().undo.position
+  }
+  **/
+
+  var transmission = {
```

Comment:
This should either be `const` or just inlined into the `return JSON.stringify()`